lib/segment/src/segment_constructor/segment_constructor_base.rs

```diff
@@ -333,40 +333,14 @@ pub(crate) fn create_vector_index(
 }
 
 pub(crate) fn create_sparse_vector_index(
-    sparse_vector_config: SparseVectorDataConfig,
-    vector_index_path: &Path,
-    id_tracker: Arc<AtomicRefCell<IdTrackerSS>>,
-    vector_storage: Arc<AtomicRefCell<VectorStorageEnum>>,
-    payload_index: Arc<AtomicRefCell<StructPayloadIndex>>,
-    stopped: &AtomicBool,
+    args: sparse_vector_index::OpenArgs,
 ) -> OperationResult<VectorIndexEnum> {
-    let vector_index = match sparse_vector_config.index.index_type {
-        SparseIndexType::MutableRam => VectorIndexEnum::SparseRam(SparseVectorIndex::open(
-            sparse_vector_config.index,
-            id_tracker.clone(),
-            vector_storage.clone(),
-            payload_index.clone(),
-            vector_index_path,
-            stopped,
-        )?),
+    let vector_index = match args.config.index_type {
+        SparseIndexType::MutableRam => VectorIndexEnum::SparseRam(SparseVectorIndex::open(args)?),
```

Comment:
Are we going to create a compressed index later? Do we want to keep both compressed and original indices in the future?

Reply:
Plan to keep those for v1.10:
- MutableRam: uncompressed since updating compressed index is costly; and `f32` only. (tho maybe we'll add f16/u8 if it would be worthy)
- Compressed immutable: Ram/Mmap; f32/f16/u8. (so, 6 combinations total)

Reply:
Got it, you did it as a next step
https://github.com/qdrant/qdrant/pull/4454/files#diff-e1db48b48e08d3c200e7f47f4080abb0316f015bdb75aedfa865e63bfaf64289:~:text=lib/segment/src/segment_constructor/segment_constructor_base.rs